pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/rule/design/xml/LawOfDemeter.xml

```diff
@@ -144,10 +155,13 @@ public class Foo {
         d.doSomethingElse();
     }
 }
+class C { void doIt() {} }
+class D { void doSomethingElse() {} }
+class Bar { C getC() {} }
         ]]></code>
     </test-code>
 
-    <test-code>
+    <test-code regressionTest="false">
         <description>Static methods</description>
         <expected-problems>1</expected-problems>
```

Comment:
I think, this should be restored - previously such things like https://github.com/spring-projects/spring-framework/tree/v5.3.13/spring-beans/src/main/java/org/springframework/beans/factory/support/AbstractBeanFactory.java#L988 have been reported as "Potential violation of Law of Demeter (static property access)"

Reply:
I think this shouldn't. The test and your example are unrelated, the test is about static methods, while the example is an access to a field of another variable. I added tests to make sure that the example is caught, but static methods should be whitelisted IMO. Calling the static method with an instance receiver is very rare and isn't worth the trouble making the rule catch it.